hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/infra/EthereumCallDataHydration.java

```diff
@@ -74,9 +74,19 @@ public HydratedEthTxData tryToHydrate(
             if (callDataFile.deleted()) {
                 return failureFrom(FILE_DELETED);
             }
+
+            final var initcodeBytes = callDataFile.contents().toByteArray();
+            // Bytes.fromHex() doesn't appreciate a leading '0x' but we supported it in mono-service
+            final String hexedInitcode;
+            if (initcodeBytes[0] == (byte) '0' && initcodeBytes[1] == (byte) 'x') {
+                hexedInitcode = new String(initcodeBytes, 2, initcodeBytes.length - 2);
+            } else {
+                hexedInitcode = new String(initcodeBytes);
+            }
```

Comment:
This snipped is taken from `HevmTransactionFactory.initcodeFor`. The idea is that we have tests the call data is prefixed with 0x. To be consistent we are removing the prefix.